collector/pkg/component/analyzer/cpuanalyzer/delete_javatrace.go

```diff
@@ -0,0 +1,75 @@
+package cpuanalyzer
+
+import (
+	"sync"
+	"time"
+)
+
+type javaTraceDeleteQueue struct {
+	queueMutex sync.Mutex
+	queue      []deleteVal
+}
+
+type deleteVal struct {
+	key       string
+	enterTime time.Time
+}
+
+func newJavaTraceDeleteQueue() *javaTraceDeleteQueue {
+	return &javaTraceDeleteQueue{queue: make([]deleteVal,0)}
+}
+
+func (dq *javaTraceDeleteQueue) GetFront() *deleteVal {
+	if len(dq.queue) > 0 {
+		return &dq.queue[0]
+	}
+	return nil
+}
+
+func (dq *javaTraceDeleteQueue) Push(elem deleteVal) {
+	dq.queue = append(dq.queue, elem)
+}
+
+func (dq *javaTraceDeleteQueue) Pop() {
+	if len(dq.queue) > 0 {
+		dq.queue = dq.queue[1:len(dq.queue)]
+	}
+}
+
+func (ca *CpuAnalyzer) JavaTraceDelete(interval time.Duration, expiredDuration time.Duration) {
+	for {
+		select {
+		case <-ca.stopProfileChan:
+			return
+		case <-time.After(interval):
+			ca.telemetry.Logger.Debug("Start regular cleaning of javatrace...")
+			now := time.Now()
+			func() {
+				ca.javaTraceExpiredQueue.queueMutex.Lock()
+				defer ca.javaTraceExpiredQueue.queueMutex.Unlock()
+				for {
+					val := ca.javaTraceExpiredQueue.GetFront()
+					if val == nil {
+						break
+					}
+					if val.enterTime.Add(expiredDuration).After(now) {
+						break
+					}
+
+					func() {
+						ca.lock.Lock()
+						defer ca.lock.Unlock()
```

Comment:
Don't use this lock for "javatrace". It is designed for "cpuevents". It may slow down or block the other goroutine. Besides, you haven't really lock the `javaTraces` map when inserting elements into it. See the method `analyzerJavaTraceTime`.